tests/test_metadata.py

```diff
@@ -262,3 +262,28 @@ def test_tdm_current_records_most_recent_version(timdex_metadata_with_deltas):
             == most_recent.iloc[0]["run_timestamp"]
         )
         assert current_version.iloc[0]["run_id"] == most_recent.iloc[0]["run_id"]
+
+
+def test_tdm_merge_append_deltas(timdex_metadata_with_deltas):
```

Comment:
Commenting here on this test, but it feels like there could be a bit more testing.

This test appears to test that, after merging, we have the same _number_ of `metadata.records` as we did before merge.  And, that the local append deltas directory is empty.

I wonder if we could also test:

- After merge, all rows that _were_ in `metadata.append_deltas` can now be found in `static_db.records`.
- If _during_ a merge a _new_ append delta is written, can we be sure we didn't delete it (because we know we would have missed inserting into the static file)?

I'm unsure if helpful, but it almost seems like you could have a fixture which is _after_ a merge as well.  Then you could compare "before" and "after" TD and TDM instances and not have to juggle variables and old counts.